File: commands/welcome.py
import discord
import requests
from discord.ext import commands


def setup(bot: discord.Bot):
    @bot.slash_command(
        name="welcome",
        #description="Resends all welcome messages of all members to a specified channel.",
        description="Command is disabled.",
        default_member_permissions=discord.Permissions(administrator=True),
    )
    @commands.is_owner()
    async def welcome_cmd(
        ctx: discord.ApplicationContext,
        channel: discord.TextChannel = discord.Option(
            discord.TextChannel, "The channel to send the messages in."
        ),
        mentions: bool = discord.Option(
            bool, "Mentions every members in the messages if true"
        ),
        welcome_message: str = discord.Option(
            str, "The welcome message", required=False, default=None
        ),
    ):
        
        #await ctx.respond("Sending welcome messages...", ephemeral=True)
        response = requests.get("https://api.uniqueweb.site/v2/botwelcome/status/disabled")
        if response.text == "true":
            await ctx.respond("Due to Top.gg's rules, the /welcome is removed for the time being.", ephemeral=True)
            return
        else:
            mention_setting = discord.AllowedMentions.all() if mentions else discord.AllowedMentions.none()

            for member in ctx.guild.members:
                await channel.send(
                    f"\nWelcome {member.mention} to **{ctx.guild.name}**!\n\nMake sure to read the rules/messages!",
                    allowed_mentions=mention_setting,
                )

File: commands/welcome.js
import {
  SlashCommandBuilder,
  PermissionFlagsBits,
  ChannelType,
  MessageFlags,
} from "discord.js";

export default {
  data: new SlashCommandBuilder()
    .setName("welcome")
    .setDescription("Command is disabled.")
    .setDefaultMemberPermissions(PermissionFlagsBits.Administrator)
    .addChannelOption((o) =>
      o
        .setName("channel")
        .setDescription("The channel to send the messages in.")
        .addChannelTypes(ChannelType.GuildText)
        .setRequired(true)
    )
    .addBooleanOption((o) =>
      o
        .setName("mentions")
        .setDescription("Mentions every members in the messages if true")
        .setRequired(true)
    )
    .addStringOption((o) =>
      o.setName("welcome_message").setDescription("The welcome message").setRequired(false)
    ),

  async execute(interaction) {
    // owner-only (mirrors @commands.is_owner())
    if (!interaction.client.ownerIds.has(interaction.user.id)) {
      await interaction.reply({
        content: "This command is owner-only.",
        flags: MessageFlags.Ephemeral,
      });
      return;
    }

    const channel = interaction.options.getChannel("channel");
    const mentions = interaction.options.getBoolean("mentions");

    // remote kill-switch: Top.gg mass-mention rules can disable this command.
    // Fail closed — if the site doesn't reply, treat the command as disabled.
    let disabled = true;
    try {
      const res = await fetch("https://api.uniqueweb.site/v2/botwelcome/status/disabled", {
        signal: AbortSignal.timeout(2000),
      });
      disabled = (await res.text()).trim() === "true";
    } catch {
      // timeout / connection refused / DNS failure all land here
      disabled = true;
    }

    if (disabled) {
      await interaction.reply({
        content: "Due to Top.gg's rules, the /welcome is removed for the time being.",
        flags: MessageFlags.Ephemeral,
      });
      return;
    }

    await interaction.reply({
      content: "Sending welcome messages...",
      flags: MessageFlags.Ephemeral,
    });

    const allowedMentions = mentions ? { parse: ["users"] } : { parse: [] };

    const members = await interaction.guild.members.fetch();
    for (const member of members.values()) {
      await channel.send({
        content: `\nWelcome ${member} to **${interaction.guild.name}**!\n\nMake sure to read the rules/messages!`,
        allowedMentions,
      });
    }
  },
};
